logstash-core/spec/support/shared_examples.rb

```diff
@@ -119,3 +119,58 @@
     expect(LogStash::Json.load(last_response.body)["path"]).not_to be_nil
   end
 end
+
+shared_examples "returns successfully without waiting" do
+
+  it 'returns status code 200' do
+    expect(response.status).to be 200
+  end
+
+  it 'returns immediately' do
+    start_time = Time.now
+    response
+    end_time = Time.now
+    expect(end_time - start_time).to be < 0.5
+  end
+end
+
+shared_examples "timed out response" do
+
+  it 'returns an error response' do
+    expect(response.body).to include(error_message)
+  end
+
+  it 'returns an 400 status' do
+    expect(response.status).to be 400
```

Comment:
~~🤔 should this be a 503?~~

> status is provided with a timeout: wait for the given status or a better one (green > yellow > red). When the timeout is reached and neither the target status nor a better one is reached: return error response and http status 503

EDIT: the example group is misnamed, but consistently validates the behaviour of a bad request.